app/routes/application.js

```diff
@@ -180,7 +180,11 @@ export default Route.extend({
       // as we're trying to render a page and remove loading
       // indicator at a same time
       later(() => {
-        if (this.loadingView) {
+        // Clear the loader only if the loader is actually present in the DOM
+        if (
+          this.loadingView &&
+          document.getElementById(this.loadingView.elementId)
```

Comment:
Do we really need DOM check here ?  I think check on `this.loadingView.elementId` is enough. Thoughts ?